server/src-lib/Hasura/RQL/DML/RemoteJoin.hs

```diff
@@ -215,14 +219,19 @@ compositeValueToJSON = \case
   CVObjectArray vals -> AO.array $ map compositeValueToJSON vals
   CVFromRemote v -> v
 
+-- | A 'RemoteJoinField' carries the minimal GraphQL AST of a remote relationship field.
+-- All such 'RemoteJoinField's of a pariticular remote schema are batched together
+-- and made GraphQL request to remote server to fetch remote join values.
 data RemoteJoinField
   = RemoteJoinField
-  { _rjfRemoteSchema :: !RemoteSchemaInfo
-  , _rjfAlias        :: !G.Alias
-  , _rjfField        :: !G.Field
-  , _rjfFieldCall    :: ![G.Name]
+  { _rjfRemoteSchema :: !RemoteSchemaInfo -- ^ The remote schema server info.
+  , _rjfAlias        :: !G.Alias -- ^ To level alias of the field
```

Comment:
"Top level"?  These comments are a huge improbement, thanks!

Reply:
Yes, It is :smile: . I fixed the typo in latest commit. Now the PR is up to date with the master branch. Awaiting for your review. Thank you.